cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1520,10 +1521,10 @@ elaborateInstallPlan verbosity platform compiler compilerprogdb pkgConfigDB
                 external_exe_dep_sids ++ external_lib_dep_sids
 
             external_exe_map = Map.fromList $
-                [ (getComponentId pkg, path)
+                [ (getComponentId pkg, paths)
                 | pkg <- external_exe_dep_pkgs
-                , Just path <- [planPackageExePath pkg] ]
-            exe_map1 = Map.union external_exe_map exe_map
+                , let paths = planPackageExePaths pkg ]
+            exe_map1 = Map.union external_exe_map (pure <$> exe_map)
```

Comment:
I'd rather be a bit more explicit here and write this as `M.map singleton exe_map` (if there's no `singleton :: a -> [a]` in `cabal-install`, we can add it to `Distribution.Compat.Prelude` or a utils module. Or just use `(\a -> [a])`.